app/helpers/katello/katello_urls_helper.rb

```diff
@@ -1,15 +1,27 @@
 require 'uri'
+
 module Katello
   module KatelloUrlsHelper
+    def foreman_settings_url
+      Setting[:foreman_url].sub(/^https/, 'http')
+    end
+
     def subscription_manager_configuration_url(host = nil, rpm = true)
       prefix = if host && host.content_source
-                 "http://#{@host.content_source.hostname}"
+                 "http://#{host.content_source.hostname}"
                else
-                 Setting[:foreman_url].sub(/\Ahttps/, 'http')
+                 foreman_settings_url
                end
       config = rpm ? SETTINGS[:katello][:consumer_cert_rpm] : SETTINGS[:katello][:consumer_cert_sh]
 
       "#{prefix}/pub/#{config}"
     end
+
+    def repository_url(content_path, schema = 'http')
+      path = ::Katello::Glue::Pulp::Repos.repo_path_from_host(@host, content_path)
```

Comment:
Do we actually need a special method for `host`. AFAIK `repo_path_from_content_path(@host.lifecycle_environment, content_path)` would work just fine, wouldn't it?

Reply:
On my tests `organisation` was not part of  `lifecycle_environment`, so I separated what enters to the method.

Have I missed something?

Reply:
fixed, following an explanation what I was missing.